Examples/Algorithms/Alignment/include/ActsExamples/Alignment/AlignmentAlgorithm.hpp

```diff
@@ -24,6 +24,54 @@
 
 namespace ActsExamples {
 
+class AlignmentGroup {
+
+public:
+    AlignmentGroup(const std::string& name, const std::vector<GeometryIdentifier>& geoIds)
+        : m_name(name), m_map(constructHierarchyMap(geoIds)) {
+    }
+
+    // Access the name of the group
+    std::string getNameOfGroup() const {
+        return m_name;
+    }
+
+private:
+    std::string m_name;  //  storing the name in the class
+    GeometryHierarchyMap<bool> m_map;
+
+    GeometryHierarchyMap<bool> constructHierarchyMap(const std::vector<GeometryIdentifier>& geoIds) {
+        std::vector<GeometryHierarchyMap<bool>::InputElement> ies;
+        for (const auto& geoId : geoIds) {
+            ies.emplace_back(geoId, true);
+        }
+        return GeometryHierarchyMap<bool>(ies);
+    }
+};
+
+class AlignmentAlgorithm {
+public:
+    void makeAlignmentGroup(const std::vector<AlignmentParameters>& alignments);
+
+private:
+    std::map<GeoIds, AlignmentGroup> alignmentGroups;  //  storage of AlignmentGroups 
+};
+
+ // function that will group al_par based on geoIds 
+
+void AlignmentAlgorithm::makeAlignmentGroup(const std::vector<AlignmentParameters>& alignments) {
+    for (const auto& alignmentPar : alignments) {
+        GeoIds superstructure_id = alignmentPar.getGeoIds();
+        if (alignmentGroups.find(superstructure_id) == alignmentGroups.end()) {
+            alignmentGroups[superstructure_id] = AlignmentGroup(superstructure_id);
+        }
+        AlignmentGroup& alignmentGroup = alignmentGroups[superstructure_id];
+        // @todo: each superstructure_id corresponds to a specific group,
+        // and for each group add a set of associated alignment parameters
+    }
+    // later: Define a function that will add a set of al_par (for each group)
+}
+
```

Comment:
I think you can remove this -- 1) the AlignmentAlgorithm is already defined below and 2) the AlignmentGroup constructor is actually our makeAlignmentFunction so we do not need a separate one after all

Reply:
Implemented.